modules/calendar/includes/get_provider_events.php

```diff
@@ -24,7 +24,35 @@
 
 $fetchedEvents = fetchAllEvents($_POST['start'], $_POST['end']);
 
+// facility ACL check
+if ($GLOBALS['facility_acl'] == 1) {
+  $facility_filter = do_action('filter_patient_select_pnuserapi', $_SESSION['authUser']); // returns schedule/allowed facility ids in a SQL string
+  $facilities_allowed_to_user = array();  // array containing facility ids which logged in user is allowed to see
+
+  // making of $facilities_allowed_to_user
+  if ($facility_filter) {
+    // $facility_filter = " pd.facility IN ( user_schedule_facility_id_string ) " from tags_filters/start.php
+    $facility_filter = substr($facility_filter, strpos($facility_filter, 'IN'));
+    $facility_filter = ' facility.id ' . $facility_filter;
+    $sql = "SELECT facility.id AS fid
+            FROM facility
+            WHERE" . $facility_filter; // $facility_filter = " facility.id IN ( user_schedule_facility_id_string ) "
```

Comment:
Boo-Boo here.  should say 
            `WHERE" . $facility_filter;"; // $facility_filter = " facility.id IN ( user_schedule_facility_id_string ) "`

Reply:
Only took me 14 hours to find a simple syntax error... 😜

Reply:
Correction:  
    $sql = "SELECT facility.id AS fid FROM facility WHERE" . $facility_filter.";"; // $facility_filter = " facility.id IN ( user_schedule_facility_id_string ) "

Reply:
That looks like it would stand a chance of working the other one did not.

So did that fix the error?

Reply:
NO.
The error log output debug string echoing the $sql reads:
get_provider_events.php query string: SELECT facility.id AS fid FROM facility WHERE  facility.id  pd.facility = '-1' ;

The concatenated value must be not right...

Reply:
Oh geezz...maybe that was NOT supposed to be a comment!

Reply:
No that is ok, Do you have a facility assigned to the user?

Reply:
Yes.
The error_log($sql) output (and subsequent syntax error generated) shows that it amounts to a bad query.
WHERE facility.id pd.facility = '-1'  isn't good.  There is no join there.  The function that it pulls that from with all the substrings and such is a bit beyond me this late at night.  I guess I should track them down to try to help Apoor when he wakes up to this mess of typing I have been generating.

Reply:
From what I have figured out the actual thing that is being returned by the code is

 pd.facility = '-1'

Which indicated there is no facility information getting rendered from the do action thingie.

function tf_filter_patient_select_pnuserapi2( $username )
{
    $facilitiesToShow = get_facilities_to_show( $username );
    $where = " pd.facility = '-1' ";
    if ( count( $facilitiesToShow ) ) {
        $facilityString = implode(",", $facilitiesToShow);
        $where = " pd.facility IN ( $facilityString ) ";
    }
    return $where;
}
add_action( 'filter_patient_select_pnuserapi', 'tf_filter_patient_select_pnuserapi2' );

Reply:
I guess the function call to start.php isn't returning an expected value.  I swear I think I have the database set up to use the feature.  This thing has been a pain for years now.  It was a pain in the old version due to non-optimal configuration and hinkey stuff, and it is a pain now.  Apoor will probably figure this out in two minutes.

Reply:
Putting `error_log($sql);` after this sql statement resulted in following logs:
1. When logged in as a user with no mention in `users_facility` table, i.e., no allowed facilities:
`SQL Error with statement:query failed: SELECT facility.id AS fid
            FROM facility
            WHERE facility.id  pd.facility = '-1' --You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'pd.facility = '-1'' at line 3==>C:\wamp64\www\pro\lh-ehr\modules\calendar\includes\get_provider_events.php at 41:sqlStatement`

2. When logged in as a user with a few mention in users_facility table, i.e., some (here 2) allowed facilities:
`SELECT facility.id AS fid
            FROM facility
            WHERE facility.id IN ( 5,4,5 )`

I think the code in start.php is working properly and returning the required string of allowed facilities and the issue may be resulting from the case when it returns `$where = " pd.facility = '-1' ";` because of having no allowed facilities corresponding to logged in user in users_facility table. 

And I haven't included any handling of this case in my code resulting in sql error. I'll add it in my next commit. @aethelwulffe When logged in as a particular user, please check if that user has at least 1 corresponding facility in users_facility table and see if correct sql (with facility id string) is logged. 

In case of no allowed facilities for logged in user (when there is no mention in users_facility table), I think there is no point of performing any sql query because `$facilities_allowed_to_user` would be empty even if we perform query by changing `pd.facility = '-1'` to `facility.id = '-1'`. And all events in Calendar would not be fetched/displayed.